app/services/conviction_decision_tree.rb

```diff
@@ -45,7 +45,7 @@ def destination
   def after_conviction_subtype
     return edit(:conviction_bail)    if conviction_subtype.bailable_offense?
     return edit(:compensation_paid)  if conviction_subtype.compensation?
-    return after_motoring_conviction if conviction_type.motoring?
+    return after_motoring_conviction if conviction_type.motoring? && !conviction_subtype.motoring_penalty_points?
```

Comment:
I think it might be more clear if we leave the condition as it was, and add the different branching in the `after_motoring_conviction` method. This method is becoming a bit messy with so many conditions.

Like this:

```ruby
  def after_motoring_conviction
    return edit(:motoring_endorsement) unless conviction_subtype.motoring_penalty_points?

    known_date_question
  end
```

Reply:
Looks good. I've updated this file with this suggestion.